pkg/cmd/bootstrap.go

```diff
@@ -123,6 +125,34 @@ func nonInteractiveMode(io *BootstrapParameters, client *utility.Client) error {
 	if err := checkMandatoryFlags(mandatoryFlags); err != nil {
 		return err
 	}
+	if io.GitHostAccessToken != "" {
+		err := setSecretIfNotSet(io.GitOpsRepoURL, io.GitHostAccessToken)
+		if err != nil {
+			return err
+		}
+		err = setSecretIfNotSet(io.ServiceRepoURL, io.GitHostAccessToken)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func setSecretIfNotSet(repoURL, accessToken string) error {
+	hostName, err := webhook.HostFromURL(repoURL)
```

Comment:
I'm not in favor of using hostname.  The reason is that we can't assume one token for all repos of the same git provider.   I can be given a token to setup webhook on a particular repo which my account does not have permission to.   It actually happnens to me at Red Hat when I set up openshift-ci for this kam repo.   But if you insist, then it's fine with me.